src/funmisc.c

```diff
@@ -1562,3 +1567,273 @@ FUNCTION(fun_benchmark)
+      snprintf(ct_header, sizeof ct_header, "Content-Type: %s", contenttype);
+      headers = curl_slist_append(headers, ct_header);
+    }
+    if (contenttype && postdata &&
+        (strstr(contenttype, "charset=utf-8") ||
+         strstr(contenttype, "charset=UTF-8"))) {
+      int ulen;
+      char *utf8 = latin1_to_utf8(args[3], strlen(args[3]), &ulen, 0);
+      curl_easy_setopt(handle, CURLOPT_COPYPOSTFIELDS, utf8);
+      mush_free(utf8, "string");
+    } else if (postdata) {
+      curl_easy_setopt(handle, CURLOPT_COPYPOSTFIELDS, args[3]);
+    }
+  }
+
+  curl_easy_setopt(handle, CURLOPT_PRIVATE, req);
+
+  headers = curl_slist_append(headers, "Accept-Charset: iso-8859-1, utf-8, us-ascii");
+  req->header_slist = headers;
+  curl_easy_setopt(handle, CURLOPT_HTTPHEADER, headers);
+
+  ncurl_queries += 1;
+
+  curl_status = curl_easy_perform(handle);
+  
+  if (curl_status == CURLM_OK) {
+    long respcode;
+    char *contenttype;
+    struct urlreq *resp;
+    bool is_utf8 = 0;
+
+    curl_easy_getinfo(handle, CURLINFO_PRIVATE, &resp);
+
+    if (curl_status == CURLE_OK || resp->too_big) {
+      if (curl_easy_getinfo(handle, CURLINFO_RESPONSE_CODE, &respcode) ==
+          CURLE_OK) {
+        if (respcode) {
+          pe_regs_set_int(resp->pe_regs, PE_REGS_Q, "status", respcode);
+        }
+      }
+      if (curl_easy_getinfo(handle, CURLINFO_CONTENT_TYPE, &contenttype) ==
+          CURLE_OK) {
+        if (contenttype) {
+          pe_regs_set(resp->pe_regs, PE_REGS_Q, "content-type", contenttype);
+          if (strstr(contenttype, "charset=utf-8") ||
+              strstr(contenttype, "charset=UTF-8")) {
+            is_utf8 = 1;
+          }
+        }
+      }
+      if (resp->body && sqlite3_str_length(resp->body) > 0) {
+        char rbuff[BUFFER_LEN];
+        char *body = NULL;
+        char *latin1 = NULL;
+        int len;
+        int body_size = sqlite3_str_length(resp->body);
+        body = sqlite3_str_finish(resp->body);
+        resp->body = NULL;
+        if (is_utf8) {
+          latin1 = utf8_to_latin1(body, body_size, &len, 1, "string");
+          if (len >= BUFFER_LEN) {
+            resp->too_big = 1;
+            latin1[BUFFER_LEN - 1] = '\0';
+          }
+        } else {
+          latin1 = body;
+          if (body_size >= BUFFER_LEN) {
+            resp->too_big = 1;
+            body[BUFFER_LEN - 1] = '\0';
+          }
+        }
+        pe_regs_setenv(resp->pe_regs, 0, latin1);
+
+        call_ufun(&ufun, rbuff, executor, enactor, pe_info, resp->pe_regs);
+        safe_str(rbuff, buff, bp);
+        
+        if (is_utf8) {
+          mush_free(latin1, "string");
+        }
+        if (body) {
+          sqlite3_free(body);
+        }
+      }
+      if (resp->too_big) {
+        notify(resp->thing, "Too much HTTP data received; excess truncated.");
+      }
+    } else {
+      pe_regs_set(pe_info->regvals, PE_REGS_Q, "http-error", curl_easy_strerror(curl_status));
```

Comment:
I am using the `http-error` register here because it would be a bad time if it returned a string containing the error at this point.